StoredProcedureEFCore/OutputParameter.cs

```diff
@@ -3,17 +3,31 @@
 
 namespace StoredProcedureEFCore
 {
-  internal class OutputParam<T> : IOutParam<T>
-  {
-    public OutputParam(DbParameter param)
+    internal class OutputParam<T> : IOutParam<T>
     {
-      _dbParam = param;
-    }
+        public OutputParam(DbParameter param)
+        {
+            _dbParam = param;
+        }
+
+        public T Value
+        {
+            get
+            {
+                if (_dbParam.Value is System.DBNull)
+                {
+                    if (Nullable.GetUnderlyingType(typeof(T)) != null)
```

Comment:
You can put this check in the first if

Reply:
The reference check is now implicit. I have added the exception. I have expanded the correxponding tests.